lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -193,6 +193,67 @@ def wire_target_dependencies(target_installation_results)
           PodTargetDependencyInstaller.new(sandbox, pod_target_installation_results_hash, metadata_cache).install!
         end
 
+        def _merge_spm_requirements(url, requirements)
+          if requirements.count > 1
+            requirements.reduce(requirements.first) do |acc, req|
+              if acc != req
+                error "SPM dependencies with different requirements: #{requirements} for #{url} is not yet supported"
+              end
+              acc
+            end
+          else
+            requirements.first
+          end
+        end
+
+        def install_spm_dedpendecies(project, pod_targets)
+          spm_dependencies = pod_targets.flat_map { |target| target.root_spec.consumer(target.platform).spm_dependencies }
+          spm_dependencies_by_url = spm_dependencies.group_by { |dep| dep[:url] }
+          merged_spm_dependencies = spm_dependencies_by_url.map do |url, deps|
+            [url, _merge_spm_requirements(url, deps.map { |dep| dep[:requirement] })]
+          end
+
+          spm_dependencies_by_url = merged_spm_dependencies.map do |url, requirement|
+            [url, project.add_spm_dependency(url, requirement.to_h)]
+          end
+
+          spm_dependencies_by_url.to_h
+        end
+
+        def _spm_deps_workaround_for_swift(native_target)
+          # Seems to be an XCode issue where subprojects SWITFT_INCLUDE_PATHS doesn't the added SPM dependency's products
+          # without this Swift code trying to import the SPM dependency's product will fail with "No such module"
+          search_path = '${SYMROOT}/${CONFIGURATION}${EFFECTIVE_PLATFORM_NAME}/'
+          native_target.build_configurations.each do |config|
+            build_settings = native_target.build_settings(config.name)
+            build_settings['SWIFT_INCLUDE_PATHS'] ||= ['$(inherited)']
+            unless build_settings['SWIFT_INCLUDE_PATHS'].include?(search_path)
+              build_settings['SWIFT_INCLUDE_PATHS'].push(search_path)
+            end
+          end
+        end
+
+        def wire_target_spm_dependencies(target_installation_results, spm_dependencies_by_url)
+          target_installation_results.pod_target_installation_results.each_value do |_, pod_target_installation_result|
```

Comment:
Hi, 
Thank you for this PR, 
I tested it and found out that it wasn't working in my case, I had to change  this line to make it work 
```suggestion
          target_installation_results.pod_target_installation_results.each_value do |pod_target_installation_result|
```
I had error telling me that `pod_target_installation_result` was nil without this change.

### Error

```
NoMethodError - undefined method `native_target' for nil:NilClass
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer/xcode/pods_project_generator.rb:246:in `block (2 levels) in wire_target_spm_dependencies'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer/xcode/pods_project_generator.rb:245:in `each'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer/xcode/pods_project_generator.rb:245:in `block in wire_target_spm_dependencies'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer/xcode/pods_project_generator.rb:237:in `each_value'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer/xcode/pods_project_generator.rb:237:in `wire_target_spm_dependencies'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer/xcode/single_pods_project_generator.rb:26:in `generate!'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer.rb:320:in `block in create_and_save_projects'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/user_interface.rb:64:in `section'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer.rb:315:in `create_and_save_projects'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer.rb:307:in `generate_pods_project'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer.rb:183:in `integrate'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/installer.rb:170:in `install!'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/command/install.rb:52:in `run'
/Users/sdrode/.rvm/gems/ruby-3.0.0/bundler/gems/CLAide-97b765ec88aa/lib/claide/command.rb:334:in `run'
/Users/sdrode/Downloads/CocoaPods/lib/cocoapods/command.rb:52:in `run'
/Users/sdrode/Downloads/CocoaPods/bin/pod:55:in `<main>'
```

I'm no Ruby expert so maybe I didn't understand something 🙂

Reply:
@Dragna thanks much, sorry for the issue and thanks for checking out and the patch. I've now updated my PR.